server/src/main/java/org/apache/druid/indexing/overlord/supervisor/SupervisorStateManagerConfig.java

```diff
@@ -46,6 +46,12 @@
   @JsonProperty
   private int maxStoredExceptionEvents = Math.max(unhealthinessThreshold, healthinessThreshold);
 
+  @JsonProperty("idleConfig.enabled")
+  private boolean idleConfigEnabled = false;
```

Comment:
Maybe instead of flattening the spec, keep an `idleConfig` itself here and handle it correctly inside the `SeekableStreamSupervisor`.

Reply:
We're passing 2 spec mappings `druid.supervisor.idleConfig.enabled, druid.supervisor.idleConfig.inactiveAfterMillis` rather than an idleConfig json object, I am not sure if it can be deserialised to idleConfig mapping.